include/type_safe/boolean.hpp

```diff
@@ -210,4 +225,11 @@ namespace type_safe
 #undef TYPE_SAFE_DETAIL_MAKE_PREDICATE
 } // namespace type_safe
 
+namespace std{
+	size_t hash<type_safe::boolean>::operator()(const type_safe::boolean& boolean) const
+	{
+		return hash<bool>()(boolean.value_);
```

Comment:
Instead of accessing value_, you can static_cast the boolean to bool. Then you also don't need the friend and can put the entire hash struct here without the declaration up front.